text/0000-signed-transaction-payload/0000-signed-transaction-payload.md

```diff
@@ -0,0 +1,825 @@
+    </tr>
+    <tr>
+        <td>Payload Type</td>
+        <td>varint</td>
+        <td>
+        Set to <strong>value 0</strong> to denote a <i>Signed Transaction</i> payload.
+        </td>
+    </tr>
+    <tr>
+        <td valign="top">Transaction <code>oneOf</code></td>
+        <td colspan="2">
+            <details open="true">
+                <summary>Unsigned Transaction</summary>
+                <blockquote>
+                Describes the essence data making up a transaction by defining its inputs and outputs and an optional payload.
+                </blockquote>
+                <table>
+                    <tr>
+                        <td><b>Name</b></td>
+                        <td><b>Type</b></td>
+                        <td><b>Description</b></td>
+                    </tr>
+                    <tr>
+                        <td>Transaction Type</td>
+                        <td>varint</td>
+                        <td>
+                        Set to <strong>value 0</strong> to denote an <i>Unsigned Transaction</i>.
+                        </td>
+                    </tr>
+                   <tr>
+                        <td>Inputs Count</td>
+                        <td>varint</td>
+                        <td>The amount of inputs proceeding.</td>
+                    </tr>
+                   <tr>
+                        <td valign="top">Inputs <code>anyOf</code></td>
+                        <td colspan="2">
+                            <details>
+                                <summary>UTXO Input</summary>
+                                <blockquote>
+                                Describes an input which references an unspent transaction output to consume.
+                                </blockquote>
+                                <table>
+                                    <tr>
+                                        <td><b>Name</b></td>
+                                        <td><b>Type</b></td>
+                                        <td><b>Description</b></td>
+                                    </tr>
+                                    <tr>
+                                        <td>Input Type</td>
+                                        <td>varint</td>
+                                        <td>
+                                            Set to <strong>value 0</strong> to denote an <i>UTXO Input</i>.
+                                        </td>
+                                    </tr>
+                                    <tr>
+                                        <td>Transaction ID</td>
+                                        <td>ByteArray[32]</td>
+                                        <td>The BLAKE2b-256 hash of the transaction from which the UTXO comes from.</td>
+                                    </tr>
+                                    <tr>
+                                        <td>Transaction Output Index</td>
+                                        <td>varint</td>
```

Comment:
why use varint if you set the rule to be < 127? why not uint8?

Reply:
I am afraid that 127 might be too limiting for the allowed number of inputs/outputs.

It might be worth considering a fixed integer type for this field to assure that an input block always has a fixed size.